sunpy/net/jsoc/jsoc.py

```diff
@@ -33,71 +34,40 @@ class NotExportedError(Exception):
     pass
 
 
-class JSOCResponse(BaseQueryResponse):
+class JSOCResponse(BaseQueryResponseTable):
     def __init__(self, table=None, client=None):
         """
         table : `astropy.table.Table`
         """
-        super().__init__()
-        self.table = table or astropy.table.QTable()
+        super().__init__(table, client)
         self.query_args = getattr(table, 'query_args', None)
         self.requests = getattr(table, 'requests', None)
-        self._client = client
-
-    @property
-    def client(self):
-        return self._client
-
-    @client.setter
-    def client(self, client):
-        self._client = client
-
-    @property
-    def blocks(self):
-        return list(self.table.iterrows)
-
-    def __len__(self):
-        if self.table is None:
-            return 0
-        else:
-            return len(self.table)
 
     def __getitem__(self, item):
-        if isinstance(item, int):
-            item = slice(item, item + 1)
-        ret = type(self)(self.table[item])
+        ret = super().__getitem__(item)
         ret.query_args = self.query_args
-        ret.requests = self.requests
         ret.client = self._client
-
         warnings.warn("Downloading of sliced JSOC results is not supported. "
                       "All the files present in the original response will be downloaded.",
                       SunpyUserWarning)
         return ret
 
-    def __iter__(self):
-        return (t for t in [self])
-
     def build_table(self):
-        return self.table
-
-    def append(self, table):
-        if self.table is None:
-            self.table = table
-        else:
-            self.table = astropy.table.vstack([self.table, table])
+        # remove this check post 2.1
+        if any('keys' in i for i in self.query_args):
+            return self.table
 
-    def response_block_properties(self):
-        """
-        Returns a set of class attributes on all the response blocks.
-        """
-        warnings.warn("The JSOC client does not support response block properties.", SunpyUserWarning)
-        return set()
+        default_columns = ['T_REC', 'TELESCOP', 'INSTRUME', 'WAVELNTH', 'CAR_ROT']
+        cols_in_table = list()
+        for colname in default_columns:
+            if colname in self.table.colnames:
+                cols_in_table.append(colname)
```

Comment:
Same structure of the for loop with an if.
```suggestion
        cols_in_table = [colname for colname in default_columns if colname in self.table.colnames]
```